src/all/pandachaika/src/eu/kanade/tachiyomi/extension/all/pandachaika/PandaChaika.kt

```diff
@@ -0,0 +1,303 @@
+        val parodies = tags.filter { it.startsWith("parody:") }.emptyToNull()?.joinToString { it.substringAfter(":").replace("_", " ") }
+        title = this@toSManga.title
+        url = download.substringBefore("/download/")
+        author = (groups ?: artists)
+        artist = artists
+        genre = listOfNotNull(male, female, others).joinToString().takeIf { it.isNotEmpty() }
+        description = buildString {
+            append("Uploader: ", uploader, "\n")
+            publishers?.let {
+                append("Publishers: ", it, "\n")
+            }
+            parodies?.let {
+                append("Parodies: ", it, "\n")
+            }
+            male?.let {
+                append("Male Tags: ", it, "\n")
+            }
+            female?.let {
+                append("Female Tags: ", it, "\n")
+            }
+            others?.let {
+                append("Other Tags: ", it, "\n\n")
+            }
+
+            title_jpn?.let { append("Japanese Title: ", it, "\n") }
+            append("Pages: ", filecount, "\n")
+
+            try {
+                append("Posted: ", dateReformat.format(Date(posted * 1000)), "\n")
+            } catch (_: Exception) {}
+        }
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        initialized = true
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        return response.parseAs<Archive>().toSManga()
+    }
+
+    // Chapters
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val archive = response.parseAs<Archive>()
+
+        return listOf(
+            SChapter.create().apply {
+                name = "Chapter"
+                url = archive.download.substringBefore("/download/")
+                date_upload = archive.posted * 1000
+            },
+        )
+    }
+
+    override fun getMangaUrl(manga: SManga) = baseUrl + manga.url
+    override fun getChapterUrl(chapter: SChapter) = baseUrl + chapter.url
+
+    override fun imageRequest(page: Page): Request {
+        val remoteZip = page.imageUrl!!.substringAfter("and then this:").parseAs<RemoteZip>()
+        val filename = page.imageUrl!!.substringBefore("and then this:")
+
+        val uncompressedBytes = remoteZip.fetch(filename)
+        val imageUrl = Base64.encodeToString(uncompressedBytes, Base64.DEFAULT)
+
+        var type = filename.substringAfterLast('.').lowercase()
+        type = if (type == "jpg") "jpeg" else type
+
+        return GET("https://127.0.0.1/?images/$type;base64,$imageUrl", headers)
+    }
+
+    // Pages
+    override fun pageListParse(response: Response): List<Page> {
+        fun List<String>.sort() = this.sortedBy { it.filter(Char::isDigit).toIntOrNull() }
```

Comment:
would this perform a natural sort on filenames?

Reply:
It should sort the list based on the numbers in their filenames, as most filenames usually include page numbers.

Reply:
They also have a website section that specifies the file index alongside the filename, like [here](https://panda.chaika.moe/archive/53071/image-data-list/). 
It's possible to use that to sort the filenames, but the current sort is simpler.

Reply:
I meant sorting like 1, 2, 3... instead of 10, 1, 2 ... like how normally computers sort. but if it works correctly then it's ok.

Reply:
![image](https://github.com/keiyoushi/extensions-source/assets/48650614/9c545ab4-0c24-4f10-aa0b-883b404911f5)
I belive this kind of number will mess with sorting?

Reply:
Thank you. I will change it to your suggestion 🙏